plugins/kubernetes-backend/src/service/KubernetesProxy.ts

```diff
@@ -0,0 +1,221 @@
+/*
+ * Copyright 2022 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import {
+  AuthenticationError,
+  ConflictError,
+  ForwardedError,
+  InputError,
+  NotFoundError,
+} from '@backstage/errors';
+import { bufferFromFileOrString, KubeConfig } from '@kubernetes/client-node';
+import * as https from 'https';
+import fetch, { RequestInit, Response } from 'node-fetch';
+import { Logger } from 'winston';
+
+import { ClusterDetails, KubernetesClustersSupplier } from '../types/types';
+
+import type { Request as ExpressRequest, RequestHandler } from 'express';
+
+/**
+ *
+ * @alpha
+ */
+export const APPLICATION_JSON: string = 'application/json';
+
+/**
+ *
+ * @alpha
+ */
+export const HEADER_KUBERNETES_CLUSTER: string = 'X-Kubernetes-Cluster';
+
+const HEADER_CONTENT_TYPE: string = 'Content-Type';
+
+const CLUSTER_USER_NAME: string = 'backstage';
+
+/**
+ *
+ * @alpha
+ */
+export class KubernetesProxy {
+  private _clustersSupplier?: KubernetesClustersSupplier;
+
+  static readonly PROXY_PATH: string = '/proxy/:path(*)';
+
+  constructor(protected readonly logger: Logger) {}
+
+  public proxyRequestHandler: RequestHandler = async (req, res) => {
+    const requestedCluster = this.getKubernetesRequestedCluster(req);
+
+    const clusterDetails = await this.getClusterDetails(requestedCluster);
+
+    const response = await this.makeRequestToCluster(clusterDetails, req);
+
+    const contentType = req.header(HEADER_CONTENT_TYPE) || APPLICATION_JSON;
+
+    const data = contentType.includes(APPLICATION_JSON)
+      ? await response.json()
+      : await response.text();
+
+    res.status(response.status).json(data);
```

Comment:
I don't think we need to concern ourselves with parsing the proxy request or coercing it to JSON
```suggestion
    const data = await response.text();

    res.status(response.status).send(data);
```